src/app/clusters/bindings/BindingManager.cpp

```diff
@@ -49,6 +72,28 @@ class BindingFabricTableDelegate : public chip::FabricTableDelegate
 
 BindingFabricTableDelegate gFabricTableDelegate;
 
+void LoadBindingTableFromStorage(chip::Server * appServer)
+{
+    EmberBindingTableEntry * table = static_cast<EmberBindingTableEntry *>(
+        chip::Platform::MemoryAlloc(EMBER_BINDING_TABLE_SIZE * sizeof(EmberBindingTableEntry)));
+    uint16_t readSize = EMBER_BINDING_TABLE_SIZE * sizeof(EmberBindingTableEntry);
+
+    if (appServer->GetPersistentStorage().SyncGetKeyValue(chip::kBindingStoargeKey, table, readSize) == CHIP_NO_ERROR &&
```

Comment:
Direct binary representation storage of complex data structures like binding tables is not safe across versions and updates to a data structure (i.e. a future or reverted version could read garbage due to making assumptions about exact flat sizing of native structs). All other modules that do persistent storage (except FabricTable which still needs to be converted) use structured storage that can detect new fields and be adapted to versioning easily.